ci/main.go

```diff
@@ -44,10 +50,10 @@ func mainAction(cctx *cli.Context) (rerr error) {
 		return fmt.Errorf("failed to determine GOPATH: %w", err)
 	}
 	goPath := strings.TrimSpace(string(rawGoPath))
-	if goPath != "" {
+	if goPath != "" && useHostCache {
```

Comment:
The problem with host-level caching is that re-using the GOPATH from the host might end up transfering multiple GiB into the engine. This takes far longer than any benefit that might come out of it. On a CI platform like GitHub or Drone this still makes sense, though, so it was moved behind a flag that is enabled during CI.